### PR TITLE
.github/labeler: quote rules

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR fixes the Kconfig minimatch pattern added in #16518 otherwise the automatic labeler action is failing in master ATM.

### Testing procedure

Needs to be merged in master first unfortunately

### Issues/PRs references

Fixes an issue introduced by #16518

### CODE REVIEW

**Comment by aabadie** [comment]
See https://github.com/RIOT-OS/RIOT/actions/workflows/labeler.yml

File: .github/labeler.yml [hunk 1]
```diff
@@ -45,7 +45,7 @@
   - dist/tools/kconfiglib/**/*
   - makefiles/kconfig.mk
   - kconfigs/*
-  - **/Kconfig
+  - ./**/Kconfig
```
Comment (leandrolanzieri) [inline .github/labeler.yml:48]: I see that it is currently failing, but is this going to match all Kconfig files? I'm testing the glob patterns here https://globster.xyz/ is this correct @aabadie ?

Comment (aabadie) [reply .github/labeler.yml:48]: > is this going to match all Kconfig files?

Probably but I'm not 100% sure. `./` match the current directory then `**/Kconfig` says all Kconfig files in all subdirectories.

Comment (leandrolanzieri) [reply .github/labeler.yml:48]: I checked here now http://pthrasher.github.io/minimatch-test/, which is an online tester for the minimatch library but it does not seem to hit

Comment (leandrolanzieri) [reply .github/labeler.yml:48]: It actually seems like the YAML parser is interpreting the first * as an alias, perhaps we should quote the string?

Comment (aabadie) [reply .github/labeler.yml:48]: You are probably right. I'll do some tests on my personal fork.

**Comment by aabadie** [comment]
Quoting the rules works. I adapted this PR by quoting all rules. The doc rules and build system rules are also updated (the build system was also not working for the same reason).

I opened https://github.com/aabadie/RIOT/pull/29, https://github.com/aabadie/RIOT/pull/28 and https://github.com/aabadie/RIOT/pull/27 on my fork to validate things are working. Not all rules are tested but at least the `**/<file pattern>` are working with some previous rules (boards, pkg). So this change should work as before with the fix for Kconfig.

**Comment by leandrolanzieri** [comment]
> Quoting the rules works. I adapted this PR by quoting all rules. The doc rules and build system rules are also updated (the build system was also not working for the same reason).
> 
> I opened [aabadie#29](https://github.com/aabadie/RIOT/pull/29), [aabadie#28](https://github.com/aabadie/RIOT/pull/28) and [aabadie#27](https://github.com/aabadie/RIOT/pull/27) on my fork to validate things are working. Not all rules are tested but at least the `**/<file pattern>` are working with some previous rules (boards, pkg). So this change should work as before with the fix for Kconfig.

Great! Thanks for those tests!

**Review by leandrolanzieri — APPROVED** [review]
This fixes the issues with the YAML parser. ACK!

**Pull request merged**